### PR TITLE
[cli-kit] add graphqlRequest as public export and remove oxygenRequest

### PR DESCRIPTION
**Pull request by vincentezw**
### WHY are these changes introduced?
Currently cli-kit exports two functions `oxygenRequest` and `uploadOxygenDeploymentFile`. These are remnants and aren't functional.

However we are looking to create a plugin to enable deployments to Oxygen. Since Oxygen logic is not likely to be re-used by other plugins this logic can better live in our plugin. This PR removes those two functions.

In its place it exports the `graphqlRequest` function instead, which will allow us to use this in our plugin without having to load external modules. We will require some custom headers, so the function now accepts a header object rather than a token string (previously the function called `buildHeaders` with that token).

I also added a test for it.

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/cli-kit/src/public/node/api/admin.ts [hunk 1]
```diff
@@ -18,7 +19,7 @@ export async function adminRequest<T>(query: string, session: AdminSession, vari
   const api = 'Admin'
   const version = await fetchApiVersion(session)
   const url = adminUrl(session.storeFqdn, version)
-  return graphqlRequest(query, api, url, session.token, variables)
+  return graphqlRequest(query, api, url, buildHeaders(session.token), variables)
```
Comment (isaacroldan) [inline packages/cli-kit/src/public/node/api/admin.ts:22]: All graphql calls should include the default headers, so I think it'd be better for this to be inside `graphqlRequest`.
Is there a reason why oxygen wouldn't use `buildHeaders`?

File: packages/cli-kit/src/public/node/api/graphql.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * @param options - GraphQL response options.
+ */
+export async function graphqlRequest<T>(
+  query: RequestDocument,
+  api: string,
+  url: string,
+  headers: {[header: string]: string},
+  variables?: Variables,
+  options?: GraphQLResponseOptions<T>,
+): Promise<T> {
```
[the remark this answers is not in the document]
Comment (vincentezw) [reply packages/cli-kit/src/public/node/api/graphql.ts]: great suggestion, added this

File: packages/cli-kit/src/public/node/api/admin.ts (hunk 1 above)
Comment (vincentezw) [reply packages/cli-kit/src/public/node/api/admin.ts]: Oxygen-hub requires a header `X-Oxygen-Namespace-Name` which contains the namespace for the app that's deploying. For Hydrogen storefronts that would be a static value, but that may change should we allow non-storefront deployments. I think it's best if we have control over this header from our plugin.

Comment (vincentezw) [reply packages/cli-kit/src/public/node/api/admin.ts]: that makes perfect sense. I've added an optional `addedHeaders` argument 👍

**Review by isaacroldan — APPROVED** [review]

**Pull request merged**